src/window.h

```diff
@@ -0,0 +1,30 @@
+#include <SFML/Graphics.hpp>
+
+#include "bird.h"
+#include "pipes.h"
+
+#ifndef WINDOW_DEF
+#define WINDOW_DEF
+
+class Window
+{
+public:
+    Window(int screenWidth, int screenHeight, int frameRate);
+
+    void handleEvents();
+    bool isOpen();
+
+    void update();
+    void reset();
+private:
+    sf::RenderWindow window;
+
+    float frameRate;
+    sf::Time frameTime;
+    sf::Clock clk;
+
+    Bird *bird;
+    Pipes *pipes;
```

Comment:
In terms of writing performant c++ code, there is literally no good reason for the pipes or the bird to be heap allocated pointers. You end up with bad cache efficiency, extra expensive heap allocations, etc. for no reason. The code will work exactly the same way if you had a `Bird bird` member without a pointer.

You can use c++ member initializer lists to forward the values in the Window constructor to the constructor of these members as well.

```cpp
Window::Window(int screenWidth, int screenHeight, int frameRate)
    : bird(screenWidth, screenHeight, frameRate)
    , pipes(screenWidth, screenHeight, frameRate)
{
}
```